### PR TITLE
fix: Label on Artifact GC Task no longer exceeds max characters

### LINKED ISSUE #9628: artifactgc not working because of labels
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I can confirm the issues exists when I tested with `:latest`
- [ ] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/master/docs/CONTRIBUTING.md))

### What happened/what you expected to happen?

expected artifact to be gc'd

but see "Logs from the workflow controller" below, nothing on the UI indicates issue except that the workflow stays in 'workflows 
 UI' instead of moving to 'archived workflows UI', also the artifact still remains on s3

### Version

3.4.0

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
n/a
```

### Logs from the workflow controller

kubectl logs -n argo deploy/workflow-controller | grep ${workflow}

`time="2022-09-20T04:34:37.029Z" level=error msg="failed to GC artifacts" error="failed to Create WorkflowArtifactGCTask '' for Garbage Collection: WorkflowArtifactGCTask.argoproj.io \"bitbucket-workflow-20220920t1414421000pbjt2-artgc-wfcomp-2166136261-0\" is invalid: metadata.labels: Invalid value: \"bitbucket-workflow-20220920t1414421000pbjt2-artgc-wfcomp-2166136261\": must be no more than 63 characters" namespace=auth workflow=bitbucket-workflow-20220920t1414421000pbjt2
`

### Logs from in your workflow's wait container

kubectl logs -c wait -l workflows.argoproj.io/workflow=${workflow},workflow.argoproj.io/phase!=Succeeded

### PR DESCRIPTION
**Pull request by juliev0**
Fixes #9628 

This is the label that the Controller puts on the Artifact GC Tasks and then passes as an Environment Variable to the Artifact GC Pod so that it can identify the Tasks. The only requirement really is that it's unique for each GC Pod, so by hashing the pod name it's at least as unique as any 32 bit hash.

### CODE REVIEW

**Review by tooptoop4 — APPROVED** [review]

**Comment by juliev0** [comment]
> @juliev0 Can you add test for this?

Do you just mean a unit test which confirms that the label is < 63 characters?
(We do already have end to end tests to confirm that Artifact GC is working, which are passing)

File: test/e2e/testdata/artifactgc/artgc-multi-strategy-multi-anno.yaml [hunk 1]
```diff
@@ -1,9 +1,7 @@
 apiVersion: argoproj.io/v1alpha1
 kind: Workflow
 metadata:
-  generateName: two-artgc-
-  #finalizers:
-  #  - "blah"
+  generateName: testing-on-completion-and-on-deletion-strategies-
```
Comment (juliev0) [inline test/e2e/testdata/artifactgc/artgc-multi-strategy-multi-anno.yaml:4]: made this workflow name really long as a way of testing this as part of the e2e test. I confirmed it fails on master and passes here.

**Comment by juliev0** [comment]
> @juliev0 Can you add test for this?

Done. I modified the e2e test workflow to make the workflow name really long. I confirmed it fails on master and passes here.

**Review by sarabala1979 — APPROVED** [review]

**Comment by tooptoop4** [comment]
@sarabala1979 can u pls merge this?

**Pull request merged**